packages/design-system-scripts/gulp/docs/docs.js

```diff
@@ -0,0 +1,103 @@
+const streamPromise = require('../common/streamPromise');
+const { CORE_PACKAGE_NAME } = require('../common/constants');
+const { last } = require('lodash');
+const { logTask } = require('../common/logUtil');
+
+const reactDataDirectory = `tmp/data`;
+const reactDataFilename = 'react-doc.json';
+const reactDataPath = path.resolve(reactDataDirectory, reactDataFilename);
+
+/**
+ * Parses our JSX files for relevant documentation information and stores it for
+ * our other tasks to read later
+ */
+async function extractReactDocs(sourcePackageDirs, rootPath) {
+  logTask('🌪 ', 'Generating React propType documentation and grabbing raw example code');
+
+  const sources = sourcePackageDirs.map(dir => [`${dir}/src`]);
+  const sourcesGlob = `{${sources.join(',')}}`;
+
+  return streamPromise(
+    gulp
+      .src([`${sourcesGlob}/**/*.jsx`, `!${sourcesGlob}/**/*.test.jsx`])
+      .pipe(parseReactFile(rootPath))
+      .pipe(merge({ fileName: reactDataFilename }))
+      .pipe(gulp.dest(reactDataDirectory))
+  );
+}
+
+/**
+ * Copies all the fonts and images from the source package and the core design system package
+ */
+function copySourcePackageAssets(sourcePackageDir, docsPackageDir) {
+  logTask('🏞 ', `Copying fonts and images from source package into ${docsPackageDir}/dist`);
+  return copyAssets(sourcePackageDir, docsPackageDir);
+}
+
+/**
+ * Copies all the images from our docs packages
+ */
+function copyDocsPackageImages(tempDir, docsPackageDir) {
+  logTask('🏞 ', `Copying images from docs packages into ${docsPackageDir}/dist`);
+  return copyDir(`${tempDir}/src/images`, `${docsPackageDir}/dist/images`);
+}
+
+module.exports = {
+  /**
+   * Builds the docs site
+   *
+   * Note that the source package must be built before this in order to ensure
+   * that the documentation reflects the most recent version of the source.
+   */
+  async buildDocs(sourcePackageDir, docsPackageDirs, options) {
+    let message = 'Starting the documentation generation task';
+    if (options.rootPath !== '') {
+      message += ` with a root path of ${options.rootPath}`;
+    }
+    logTask('🏃 ', message);
+
+    const pkg = await getPackageJson(sourcePackageDir);
+
+    if (!options.githubUrl) {
+      options.githubUrl = `https://github.com/${pkg.repository}`;
+    }
+
+    // This is the docs package that we will receive the output dist folder.
+    const docsPackageDir = last(docsPackageDirs);
+    const sourcePackageDirs =
+      pkg.name === CORE_PACKAGE_NAME
+        ? [sourcePackageDir]
+        : [sourcePackageDir, `node_modules/${CORE_PACKAGE_NAME}`];
+
+    // Consolidate the chain of docs directories into one temporary working directory,
+    // then perform all our operations on that working directory as if it were
+    // our docs package. This allows us to have child design systems with their
+    // own docs that inherit all the docs from the core design system.
+    const tempDir = await copyDocsToTempDir(docsPackageDirs);
```

Comment:
This is an important part. Instead of building everything directly from the docs package, we copy the parent docs package and child design system docs package into a working directory first.

Reply:
I think in our existing doc script we just process both directories and then merge the pages using `uniquePages`. Is there a particular benefit for a temp directory?

Reply:
Ah, I didn't pay attention to the `uniquePages` part. This may be unnecessary then if we modify `uniquePages` to have the new desired behavior that maintains the override order

Reply:
Maintaining the original src paths might end up being the better option for link generation